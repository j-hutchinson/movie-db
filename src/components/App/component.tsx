import axios from 'axios';
import styled from 'styled-components';
import { useEffect, useState } from 'react';
import { SearchField } from '../SearchField';
import { Table } from '../Table';
import { Pagination } from '../Pagination';

const StyledContainer = styled.div`
  max-width: 800px;
  margin: 72px auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 24px;
  align-items: center;
`;

const URL = 'https://api.themoviedb.org/3/search/movie';
const API_KEY = 'XXX';

const App = () => {
  const [data, setData] = useState<any>([]);
  const [searchTerm, setSearchTerm] = useState<string>('transformers');
  const [page, setPage] = useState<number>(1);
  const [totalPages, setTotalPage] = useState<number>(1);

  useEffect(() => {
    (async function () {
      try {
        const fetchedData = await axios({
          method: 'get',
          url: `${URL}?api_key=${API_KEY}&query=${searchTerm}&page=${page}`
        });
        setData(fetchedData?.data.results);
        setTotalPage(fetchedData?.data.total_pages);
      } catch (e) {
        console.log(e);
      }
    })();
  }, [searchTerm, page]);

  const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setTimeout(() => {
      setSearchTerm(event.target.value);
    }, 1000);
  }

  return (
    <StyledContainer>
      <SearchField onChange={onSearchChange} />
      <Table content={data} />
      <Pagination currentPage={page} setPage={setPage} totalPages={totalPages} />
    </StyledContainer>
  );
}

export default App;