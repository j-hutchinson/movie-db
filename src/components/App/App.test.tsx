import { render } from '@testing-library/react';
import App from './component';

it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
})
