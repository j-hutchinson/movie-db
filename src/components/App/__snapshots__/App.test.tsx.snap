// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-gKXOVf jRKhSY"
  >
    <form
      class="sc-bczRLJ jFsvqI"
    >
      <label
        for="site-search"
      >
        Search the site:
      </label>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputTypeSearch css-c63i49-MuiInputBase-input-MuiInput-input"
            id="site-search"
            name="q"
            placeholder="Search for..."
            type="search"
            value=""
          />
        </div>
      </div>
    </form>
    <table
      class="sc-gsnTZi cgNRiL"
    >
      <thead
        class="sc-hKMtZM hWpcTv"
      >
        <tr
          class="sc-dkzDqf gjeuPE"
        >
          <th>
            Title
          </th>
          <th>
            The table header
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <div
      class="sc-jSMfEi fHBbmf"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Prev
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</DocumentFragment>
`;
